korangar/src/loaders/animation/mod.rs

```diff
@@ -402,8 +391,8 @@ fn calculate_translation_matrix(
 /// This function converts to the "normalized" coordinates of a frame part
 /// inside the frame's bounding rectangle, defined by the vertices [-1, 0],
 /// [-1, 2], [1, 0], and [1, 2].
-fn convert_coordinates(coordinates: Vector2<i32>, size: Vector2<i32>) -> Vector2<f32> {
-    assert!(size.x != 0 && size.y != 0);
+fn convert_coordinates(coordinates: Vector2<f32>, size: Vector2<f32>) -> Vector2<f32> {
+    assert!(size.x != 0.0 && size.y != 0.0);
     let x = (coordinates.x as f32 / size.x as f32 - 1.0 / 2.0) * 2.0;

```

Comment:
Please run clippy. I get many linter errors for your PR. Unecessary returns and also casts.